clients/python/src/tensorzero/client.py

```diff
@@ -38,18 +39,89 @@
 )
 
 
-class AsyncTensorZeroGateway:
+class BaseTensorZeroGateway:
+    def __init__(self, base_url: str):
+        self.base_url = base_url
+        self.logger = logging.getLogger(__name__)
+
+    def prepare_inference_request(
```

Comment:
`_prepare_inference_request`

Reply:
disagree with this. That's gonna mess up editor hinting and stuff if we obfuscate the function signature.